lib/index.js

```diff
@@ -1,44 +1,107 @@
 
-var app = Express();
+let server
 
-// Load the config
-Confit(Path.join(__dirname, '..', 'config')).create(function(err, config) {
-  if (err) {
-    throw err;
-  }
+function setupApp (config, appConfig) {
+  const app = express()
 
   // Make the config accessible to all routes
-  Object.defineProperty(app.locals, 'config', {value: config});
-
-  Winston.clear();
-  Winston.add(Winston.transports.Console, {timestamp: true, level: config.get('logging:level')});
+  Object.defineProperty(app.locals, 'config', { value: config })
 
-  var url = URL.resolve(config.get('dhis:url'), config.get('dhis:path'));
-  // Populate the cache on startup
-  Utils.populateCache(url);
-  // Repopulate the cache based on the cron pattern
-  new CronJob(config.get('cron:pattern'), Utils.populateCache.bind(null, url), true, config.get('cron:timezone'));
+  Winston.clear()
+  Winston.add(Winston.transports.Console, { timestamp: true, level: appConfig.get('logging:level') })
 
   // Set up the routes
-  app.get('/heartbeat', Routes.uptime);
-  app.use(Middleware.stats(config.get('statsd')));
-  app.get(config.get('dhis:path'), Routes.facilityCodeLookup);
+  app.get('/heartbeat', Routes.uptime)
+  app.use(Middleware.stats(appConfig.get('statsd')))
+
+  config.routes.forEach((route) => {
+    const url = URL.resolve(route.dhisUrl, route.dhisPath)
+
+    // Populate cache on startup
+    Utils.populateCache(url)
+    Utils.addCronJob(route.cronPattern, route.cronTimezone, url)
+    Utils.addRoute(app, route.dhisPath)
+  })
+  return app
+}
+exports.setupApp = setupApp
+
+// Load the config
+function start (callback) {
+  setupConfig((err, appConfig) => {
+    if (err) {
+      return callback(err)
+    }
+
+    const apiConf = appConfig.get('openhim')
+    const port = appConfig.get('server:port')
+
+    if (apiConf.register) {
+      medUtils.registerMediator(apiConf.api, mediatorConfig, (err) => {
+        if (err) {
+          Winston.error('Failed to register this mediator, check your config', err)
+          process.exit(1)
+        }
+        apiConf.api.urn = mediatorConfig.urn
+        medUtils.fetchConfig(apiConf.api, (err, newConfig) => {
+          Winston.info('Received initial config:', JSON.stringify(newConfig))
+          const config = newConfig
+          if (err) {
+            Winston.error('Failed to fetch initial config', err)
+            process.exit(1)
+          } else {
+            Winston.info('Successfully registered mediator!')
+            const app = setupApp(config, appConfig)
+            server = app.listen(port, () => {
+              if (apiConf.heartbeat) {
+                const configEmitter = medUtils.activateHeartbeat(apiConf.api)
+                configEmitter.on('config', (newConfig) => {
+                  Winston.info('Received updated config:', JSON.stringify(newConfig))
+
+                  // set new config for mediator
+                  Utils.updateConfig(app, config.routes, newConfig.routes)
```

Comment:
As @bausmeier mentioned, the cronjob doesnt get deleted when the config is updated, When updating the config, we need to remove the existing cronjobs